File: csci24000_spring2017_A6/driver.cpp
//I pledge that I have neither given nor received any help on this assignment.
#include "bubblesort.h"
#include "insertsort.h"
#include "sort.h"
#include <stdio.h>
#include <stdlib.h>
#include <fstream>
#include <iostream> 
#include <string> 
#include <sstream>

int main(void)
{
	sort *bubble;
	sort *insert;
	int y;
	int list[50];
	//creates strings to store user input
	std::string in;
	std::string x;
	while(1)
	{
		//menu 1
		
		int i = 0;
		std::cout<<"1.)Load List"<<std::endl;
		std::cout<<"2.)Exit"<<std::endl;
		std::cout<<"Please enter your selection:";
		std::cin>>x;
		std::cout<<"\n";
		//checks for selection
		if(x == "1")
		{
			std::ifstream inFile;
			inFile.open("data.txt");
			//load file into array
			if(inFile.is_open())
			{	
				std::string Line;
				std::stringstream conv;
				std::stringstream ss;
				std::string strNum;
			
				while(getline(inFile, Line))
				{
					ss.clear();
					ss.str("");
					ss.str(Line);
					while(getline(ss,strNum,','))
					{
						conv.clear();
						conv.str("");
						conv << strNum;
						conv >> list[i];
						i++;
					}
				}
				inFile.close();
			
			//print unsorted array
				std::cout<<"Unsorted List:";
				for(y=0;y<50;y++)
						{
							std::cout<<list[y]<<",";
						}
					std::cout<<"\b"<<" ";
					std::cout<<"\n";
				}
		else
		{
			std::cout<<"File not found"<<std::endl;
			break;
		}
			
			//second menu
			while(1)
			{
				
				std::cout<<"1.)Bubble Sort"<<std::endl;
				std::cout<<"2.)Insertion Sort"<<std::endl;
				std::cout<<"3.)Exit"<<std::endl;
				std::cin>>in;
				std::cout<<"\n";
				//runs bubble sort algorithm and prints results
				if(in == "1")
				{
					bubble = new bubblesort();
					bubble->sortlist(list);
					std::cout<<"Bubble Sorted list:";
					for(y=0;y<50;y++)
					{
						std::cout<<list[y]<<",";
					}
					std::cout<<"\b"<<" ";
					std::cout<<"\n";
					delete bubble;
					break;
				}
				//runs insertion sort algorithm and prints results
				else if(in == "2")
				{
					insert = new insertsort();
					insert->sortlist(list);
					std::cout<<"Insertion Sorted List:";
					for(y=0;y<50;y++)
					{
						std::cout<<list[y]<<",";
					}
					std::cout<<"\b"<<" ";
					std::cout<<"\n";
					delete insert;
					break;
				}
				//exits
				else if(in == "3")
				{
					x = "2";
					break;
				}
				//checks for invalid input
				else
				{
					std::cout<<"Please enter a valid option"<<std::endl;
					continue;
				}
			}
		}
		//exits second loop aka exits program
		if(x == "2")
		{
			std::cout<<"Goodbye!"<<std::endl;
			break;
		}
		//checks for invalid input
		else
		{
			std::cout<<"Please enter a valid option"<<std::endl;
			continue;
		}
	}
	
	
}

File: csci24000_spring2017_A5/Driver.java
//I pledge that I have neither given nor received any help on this assignment
import java.io.*;
import java.util.*;

public class Driver
{
	public static void main(String [] args)
	{
		Employee[] Array;
		Employee E1,E2,E3,E4;
		Array = new Employee[4];
		int hours = 0;
		int x = 0;
	//Displays first menu and checks for propper input
		while(true)
		{
			System.out.println("1.) Load Employees (From File)");
			System.out.println("2.) Exit");
			System.out.println("Enter your selection:");
			Scanner in = new Scanner(System.in);
			int io = in.nextInt();
			if(io == 1)
			{
				break;
			}
			else if(io == 2)
			{
				System.exit(0);
			}
			else
			{
				System.out.println("Please enter a valid option");
				continue;
			}
		}
		//reads line from the file and allocates the data to the propper job position
		try
		{	
		FileReader FRead = new FileReader("employees.txt");
		BufferedReader BRead = new BufferedReader(FRead);
		String line = BRead.readLine();
			while((line = BRead.readLine()) != null)
			{
					String [] toke = line.split(",");
					if(toke[0].equals("1"))
					{
						E1 = new PharmacyManager();
						E1.setEmployee(toke);
						Array[x] = E1;
					}
					if(toke[0].equals("2"))
					{
						E2 = new Pharmacist();
						E2.setEmployee(toke);
						Array[x] = E2;
					}
					if(toke[0].equals("3"))
					{
						E3 = new Technician();
						E3.setEmployee(toke);
						Array[x] = E3;
					}
					if(toke[0].equals("4"))
					{
						E4 = new SeniorTech();
						E4.setEmployee(toke);
						Array[x] = E4;	
					}
				x++;
			}
		BRead.close();
		}
		//handles exceptions
		catch(FileNotFoundException e)
		{
			e.printStackTrace();
		}
		catch(IOException e)
		{
			e.printStackTrace();
		}
		
		//prints second menu and checks for invalid input
		while(true)
		{
			System.out.println("1.)Print Employee Information");
			System.out.println("2.)Enter Hours Worked");
			System.out.println("3.)Calculate Paychecks");
			System.out.println("4.)Exit Program");
			Scanner in = new Scanner(System.in);
			int io = in.nextInt();
			//prints employees
			if(io == 1)
			{
				for(x=0;x<4;x++)
				{
					System.out.format("Employee ID: %-10s", Array[x].getID());
					System.out.format("Name: %-7s %-7s",Array[x].getFName(),Array[x].getLName());
					System.out.format("Rate: %-10.2f",Array[x].getRate());
					System.out.format("\n");
				}
			}
			//asks user to input the ammount of hours checks for valid response
			else if(io == 2)
			{
				while(true)
				{
					System.out.println("Please enter hours worked");
					Scanner hr = new Scanner(System.in);
					hours = hr.nextInt();
					if(hours > 0)
					{
						break;
					}
					else
					{
						System.out.println("Please enter a valid response");
					}
				}	
			}
			// checks if hours have been entered and if it has calculates pay
			else if(io == 3)
			{
				if(hours == 0)
				{
					System.out.println("No hours have been entered!");
				}
				else
				{
					for(x=0;x<4;x++)
					{
						float pay = hours * Array[x].getRate();
						System.out.format("ID:%-10s Check Ammount: $%.2f \n", Array[x].getID(),pay);	
					}
				}
			}
			// exits
			else if(io == 4)
			{
				System.exit(0);
			}
			else
			{
				System.out.println("Please Enter a valid option");
			}
		}
	}
}